pkg/mcs/tso/server/server.go

```diff
@@ -16,79 +16,114 @@ package server
 
 import (
 	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"net"
 	"net/http"
+	"net/url"
 	"os"
 	"os/signal"
+	"path"
+	"strconv"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"syscall"
 	"time"
 
 	grpcprometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
 	"github.com/pingcap/kvproto/pkg/diagnosticspb"
 	"github.com/pingcap/kvproto/pkg/tsopb"
 	"github.com/pingcap/log"
+	"github.com/pingcap/sysutil"
+	"github.com/soheilhy/cmux"
 	"github.com/spf13/cobra"
 	bs "github.com/tikv/pd/pkg/basicserver"
 	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/pkg/member"
+	"github.com/tikv/pd/pkg/systimemon"
 	"github.com/tikv/pd/pkg/tso"
+	"github.com/tikv/pd/pkg/utils/etcdutil"
 	"github.com/tikv/pd/pkg/utils/grpcutil"
 	"github.com/tikv/pd/pkg/utils/logutil"
 	"github.com/tikv/pd/pkg/utils/metricutil"
 	"github.com/tikv/pd/pkg/utils/tsoutil"
 	"github.com/tikv/pd/pkg/versioninfo"
 	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/pkg/types"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-// If server doesn't implement all methods of bs.Server, this line will result in a clear
-// error message like "*Server does not implement bs.Server (missing Method method)"
+const (
+	leaderTickInterval = 50 * time.Millisecond
+	// tsoRootPath for all tso servers.
+	tsoRootPath      = "/tso"
+	tsoClusterIDPath = "/tso/cluster_id"
+	// tsoKeyspaceGroupPrimaryElectionPrefix defines the key prefix for keyspace group primary election.
+	// The entire key is in the format of "/pd/<cluster-id>/microservice/tso/keyspace-group-XXXXX/primary" in which
```

Comment:
maybe `/pd/<cluster-id>/microservice/tso/<group-id>/primary` looks better in here.

Reply:
Yes, we can certainly move "keyspace-group-" from the path to make it more compact.